### PR TITLE
Major set of post-Moriond multilepton updates

### PR DESCRIPTION
**Pull request by peruzzim**
- reorganization of recleaner and lepton choice codes
- additions for EWKino and SOS analyses
- lepton MVA for SUSY
- misc updates and fixes to plotting tools (elist speedup) and samples files

Needs CERN-PH-CMG/cmg-cmssw#646.

### CODE REVIEW

File: RootTools/python/samples/samples_13TeV_RunIIFall15MiniAODv2.py [hunk 1]
```diff
@@ -128,9 +128,9 @@
 
 DYJetsToLL_M10to50 = kreator.makeMCComponent("DYJetsToLL_M10to50", "/DYJetsToLL_M-10to50_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/RunIIFall15MiniAODv2-PU25nsData2015v1_76X_mcRun2_asymptotic_v12-v1/MINIAODSIM", "CMS", ".*root", 18610)
 DYJetsToLL_M10to50_ext1 = kreator.makeMCComponent("DYJetsToLL_M10to50_ext1", "/DYJetsToLL_M-10to50_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/RunIIFall15MiniAODv2-PU25nsData2015v1_76X_mcRun2_asymptotic_v12_ext1-v1/MINIAODSIM", "CMS", ".*root", 18610)
-#DYJetsToLL_M5to50_LO = kreator.makeMCComponent("DYJetsToLL_M5to50_LO", "/DYJetsToLL_M-5to50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIISpring15MiniAODv2-74X_mcRun2_asymptotic_v2-v1/MINIAODSIM", "CMS", ".*root", 7.1310*10**4) #LO
+DYJetsToLL_M5to50_LO = kreator.makeMCComponent("DYJetsToLL_M5to50_LO", "/DYJetsToLL_M-5to50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIIFall15MiniAODv2-PU25nsData2015v1_76X_mcRun2_asymptotic_v12-v1/MINIAODSIM", "CMS", ".*root", 7160) # taken from McM: https://cms-pdmv.cern.ch/mcm/requests?produce=%2FDYJetsToLL_M-5to50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8%2FRunIISummer15GS-MCRUN2_71_V1-v1%2FGEN-SIM&page=0&shown=262271
```
Comment (gpetruc) [inline RootTools/python/samples/samples_13TeV_RunIIFall15MiniAODv2.py]: This is the old correct value, the intermediate wrong value or the new correct value equal to the old one?

File: RootTools/python/samples/samples_13TeV_RunIIFall15MiniAODv2.py [hunk 2]
```diff
@@ -128,9 +128,9 @@
...
 DYJetsToLL_M50 = kreator.makeMCComponent("DYJetsToLL_M50", "/DYJetsToLL_M-50_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/RunIIFall15MiniAODv2-PU25nsData2015v1_76X_mcRun2_asymptotic_v12-v1/MINIAODSIM", "CMS", ".*root", 2008.*3)
-DYJetsToLL_M50_LO =  kreator.makeMCComponent("DYJetsToLL_M50_LO", "/DYJetsToLL_M-50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIIFall15MiniAODv2-PU25nsData2015v1_76X_mcRun2_asymptotic_v12-v1/MINIAODSIM", "CMS", ".*root", 2008.*3)
+DYJetsToLL_M50_LO =  kreator.makeMCComponent("DYJetsToLL_M50_LO", "/DYJetsToLL_M-50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIIFall15MiniAODv2-PU25nsData2015v1_76X_mcRun2_asymptotic_v12-v1/MINIAODSIM", "CMS", ".*root",4895) # taken from McM: https://cms-pdmv.cern.ch/mcm/requests?produce=%2FDYJetsToLL_M-50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8%2FRunIISummer15GS-MCRUN2_71_V1-v1%2FGEN-SIM&page=0&shown=262271
```
Comment (gpetruc) [inline RootTools/python/samples/samples_13TeV_RunIIFall15MiniAODv2.py]: No

File: RootTools/python/samples/samples_13TeV_RunIISpring15MiniAODv2.py [hunk 3]
```diff
@@ -58,13 +58,14 @@
 TBar_tWch = kreator.makeMCComponent("TBar_tWch", "/ST_tW_antitop_5f_inclusiveDecays_13TeV-powheg-pythia8_TuneCUETP8M1/RunIISpring15MiniAODv2-74X_mcRun2_asymptotic_v2-v1/MINIAODSIM", "CMS", ".*root",35.6)   
 TBar_tWch_DS = kreator.makeMCComponent("TBar_tWch_DS", "/ST_tW_antitop_5f_DS_inclusiveDecays_13TeV-powheg-pythia8_TuneCUETP8M1/RunIISpring15MiniAODv2-74X_mcRun2_asymptotic_v2-v1/MINIAODSIM", "CMS", ".*root",35.6)
 T_tWch= kreator.makeMCComponent("T_tWch", "/ST_tW_top_5f_inclusiveDecays_13TeV-powheg-pythia8_TuneCUETP8M1/RunIISpring15MiniAODv2-74X_mcRun2_asymptotic_v2-v2/MINIAODSIM", "CMS", ".*root",35.6)
-T_tWch_DS = kreator.makeMCComponent("T_tWch_DS", "/ST_tW_top_5f_DS_inclusiveDecays_13TeV-powheg-pythia8_TuneCUETP8M1/RunIISpring15MiniAODv2-74X_mcRun2_asymptotic_v2-v1/MINIAODSIM", "CMS", ".*root",35.6)
+#T_tWch_DS = kreator.makeMCComponent("T_tWch_DS", "/ST_tW_top_5f_DS_inclusiveDecays_13TeV-powheg-pythia8_TuneCUETP8M1/RunIISpring15MiniAODv2-74X_mcRun2_asymptotic_v2-v1/MINIAODSIM", "CMS", ".*root",35.6)
```
Comment (gpetruc) [inline RootTools/python/samples/samples_13TeV_RunIISpring15MiniAODv2.py]: Why removing this one?

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py [hunk 4]
```diff
@@ -1,330 +1,250 @@
-#!/usr/bin/env python
```
Comment (gpetruc) [inline TTHAnalysis/macros/prepareEventVariablesFriendTree.py:1]: The diff of this file is completely messed up. What happened?

**Comment by gpetruc** [comment]
@mmasciov can you check it has no side-effects to MT2? (it shouldn't)

**Comment by mmasciov** [comment]
@mangano @pandolf
We'll go and test PR #46 , which I understand being the same PR for 80X.
In fact, at the end we are not using 76X for MT2

**Pull request merged**